internal/controller/components/trustyai/trustyai_controller_actions.go

```diff
@@ -71,3 +77,51 @@ func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 
 	return nil
 }
+
+func createConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	trustyai, ok := rr.Instance.(*componentApi.TrustyAI)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentApi.TrustyAI)", rr.Instance)
+	}
+
+	// Skip ConfigMap creation if no configuration is specified
+	if trustyai.Spec.Eval.LMEval.AllowCodeExecution == nil &&
+		trustyai.Spec.Eval.LMEval.AllowOnline == nil {
+		return nil
+	}
+
+	// Create extra ConfigMap for DSC configuration
+	configMap := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: gvk.ConfigMap.Version,
+			Kind:       gvk.ConfigMap.Kind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			// TrustyAI's own default ConfigMap name is "trustyai-service-operator-config"
+			Name:      "trustyai-dsc-config",
+			Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+			Labels: map[string]string{
+				labels.ODH.Component(ComponentName): labels.True,
+				labels.K8SCommon.PartOf:             ComponentName,
+				"app.opendatahub.io/config-type":    "dsc-config",
+			},
+			Annotations: map[string]string{
+				"opendatahub.io/managed-by":    "dsc-trustyai-controller",
+				"opendatahub.io/config-source": "datasciencecluster",
+			},
+		},
+		Data: make(map[string]string),
+	}
+
+	if trustyai.Spec.Eval.LMEval.AllowCodeExecution != nil {
+		configMap.Data["eval.lmeval.allowCodeExecution"] =
+			strconv.FormatBool(*trustyai.Spec.Eval.LMEval.AllowCodeExecution)
+	}
+	if trustyai.Spec.Eval.LMEval.AllowOnline != nil {
+		configMap.Data["eval.lmeval.allowOnline"] =
+			strconv.FormatBool(*trustyai.Spec.Eval.LMEval.AllowOnline)
+	}
+
+	return rr.Client.Patch(ctx, configMap, client.Apply,
```

Comment:
instead of doing a patch here, you could probably just add the ConfigMap to the ReconciliationRequest's Resources, 

```go
if err := rr.AddResources(configMap); err != nil {
    return err
}
```